Lib/test/test_monitoring.py

```diff
@@ -1698,3 +1698,31 @@ def run():
             self.assertEqual(caught, "inner")
         finally:
             sys.monitoring.set_events(TEST_TOOL, 0)
+
+
+class TestOptimizer(MonitoringTestBase, unittest.TestCase):
```

Comment:
@markshannon cc @gvanrossum 

I added a test, but not sure what you wanted :)
Without this fix, Python/instrumentation.c#L1589 will not be able to pass the assert through this test.

Reply:
Hmm,, interesting, It looks like I watched the Hallucination when I wrote the test in the first place :(
I will soon update the PR.

Reply:
See: https://github.com/python/cpython/pull/108539#issuecomment-1709150737